src/Entity/Int32EntityId.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+use InvalidArgumentException;
+
+/**
+ * Interface for EntityIds that can be converted to a unique positive 32 bit integer in the range
+ * [1..2147483647], and back from the entity type and the number. Entity types that do not meet this
```

Comment:
The integer type has 32 bits. This interface only allows us to use 31. I don't care too much about what the interface is called, but I do care that saying "this is a 32 bit int" is *not* sufficient.